core/block_dragger.js

```diff
@@ -307,6 +300,41 @@ Blockly.BlockDragger.prototype.endDrag = function(e, currentDragDeltaXY) {
   Blockly.Events.setGroup(false);
 };
 
+/**
+ * Calculates the drag delta and new location values after a block is dragged.
+ * If a block would not be moved because a drag target prevents block moves,
+ * then there is no delta, and the new location is the same as the startXY.
+ * @param {!Blockly.utils.Coordinate} currentDragDeltaXY How far the pointer has
+ *     moved from the start of the drag, in pixel units.
+ * @return {{delta: (!Blockly.utils.Coordinate|undefined), newLocation:
```

Comment:
This would be clearer as a typedef.

Reply:
I don't think a typedef gives us anything, as this type is only used once. If I had to copy this definition multiple places, a name could help, but as it is it doesn't seem useful. 

I like having this close to the function, you know exactly which two properties will be in the returned object. Moving that to the top of the file when it's only referenced once would just add a hassle, imo.